src/block_cache/block_io.c

```diff
@@ -293,8 +295,12 @@ __wt_blkcache_write(WT_SESSION_IMPL *session, WT_ITEM *buf, uint8_t *addr, size_
         compression_failed = 0;
         WT_ERR(btree->compressor->compress(btree->compressor, &session->iface, src, src_len, dst,
           dst_len, &result_len, &compression_failed));
+
+        compression_ratio = result_len / (src_len - WT_BLOCK_COMPRESS_SKIP);
```

Comment:
I am not sure about how the `compression_ratio` is calculated here.

Reply:
I think this should be moved within the clause where we store a compressed image (i.e: only record compression ration when using compression).
It seems like the right check is probably `result_len /  buf->size` if the calculation is moved after the line below that adds the `WT_BLOCK_COMPRESS_SKIP` to `result_len`.
I also get kind of confused about division of integers and what results that gives you - if it were me I'd write a little standalone test with some fixed numbers and matching variable types, to ensure the outcome of this works as expected.

It's a bit funny, because I suspect that a compression ratio of 1 (no compression) could be fairly common. e.g: data that can be compressed by 30% will take a page from 32k to 22k (2.5 allocation sizes), but wouldn't register as 2x compression ratio.
IDK how much we care, but if we are updating these stats, maybe reviewing the granularity makes sense as well?

I'm aware that I've already expanded the work involved here, so push back if the question seems onerous.